presto-main/src/test/java/com/facebook/presto/sql/planner/iterative/rule/test/TestEvaluateZeroSample.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.sql.planner.iterative.rule.test;
+
+import com.facebook.presto.sql.planner.iterative.rule.EvaluateZeroSample;
+import com.facebook.presto.sql.planner.plan.SampleNode.Type;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import org.testng.annotations.Test;
+
+import static com.facebook.presto.spi.type.BigintType.BIGINT;
+import static com.facebook.presto.sql.planner.assertions.PlanMatchPattern.values;
+import static com.facebook.presto.sql.planner.iterative.rule.test.PlanBuilder.expression;
+import static com.facebook.presto.sql.planner.iterative.rule.test.PlanBuilder.expressions;
+
+public class TestEvaluateZeroSample
+{
+    private final RuleTester tester = new RuleTester();
+
+    @Test
+    public void testDoesNotFire()
+            throws Exception
+    {
+        tester.assertThat(new EvaluateZeroSample())
+                .on(p ->
+                        p.sample(
+                                0.15,
+                                Type.BERNOULLI,
+                                p.values(p.symbol("a", BIGINT))))
+                .doesNotFire();
+    }
+
+    @Test
+    public void test()
+            throws Exception
+    {
+        tester.assertThat(new EvaluateZeroSample())
+                .on(p ->
+                        p.sample(
+                                0,
+                                Type.BERNOULLI,
+                                p.filter(
+                                        expression("b > 5"),
+                                        p.values(
+                                                ImmutableList.of(p.symbol("a", BIGINT), p.symbol("b", BIGINT)),
+                                                ImmutableList.of(
+                                                        expressions("1", "10"),
+                                                        expressions("2", "11"))))))
+                // TODO: verify contents
```

Comment:
It's needed because the pattern matcher for values node doesn't do any verification. We need to make sure it's an empty values vs one with some rows.